lib/registries/addon/branded/discover/template.hbs

```diff
@@ -1,42 +1,74 @@
-
-    <Button @type='create'>
-        {{!-- template-lint-disable no-bare-strings  --}}
-        Create
-    </Button>
-
-    <Button @type='destroy'>
-        {{!-- template-lint-disable no-bare-strings  --}}
-        Destroy
-    </Button>
-</div>
\ No newline at end of file
+{{! template-lint-disable no-implicit-this }}
+{{title (t 'registries.discover.page_title')}}
+
+<HeroOverlay local-class='DiscoverHero'>
+    {{#registries-header notBranded=false showHelp=true value=(mut query) onSearch=(action 'onSearch') searchable=searchable as |header|}}
+        {{#header.lead}}
+            <a href='https://share.osf.io/' local-class='ShareLogo' data-test-share-logo title={{t 'registries.discover.SHARE'}} onclick={{action 'click' 'link' 'Discover - SHARE Logo' target=analytics}}></a>
+        {{/header.lead}}
+    {{/registries-header}}
+</HeroOverlay>
+{{#registries-discover-search
+    results=results
+    isLoading=doSearch.isIdle
+    searchOptions=searchOptions
+    onSearchOptionsUpdated=(action onSearchOptionsUpdated)
+    as |discover|
+}}
+
+    {{#discover.sidebar filterStyles=filterStyles as |sidebar|}}
+        {{registries-provider-facet
+            options=filterableSources
+            searchOptions=searchOptions
+            onSearchOptionsUpdated=sidebar.onSearchOptionsUpdated
+        }}
+
+        {{registries-registration-type-facet
+            searchOptions=searchOptions
+            onSearchOptionsUpdated=sidebar.onSearchOptionsUpdated
+        }}
+    {{/discover.sidebar}}
+
+    {{#if (not doSearch.isIdle) }}
+        <div class='text-center p-v-md' aria-label={{t 'eosf.components.discoverPage.searchLoading'}}>
+            {{loading-indicator dark=true}}
+        </div>
+    {{else}}
+        {{registries-discover-results-header
+            totalResults=totalResults
+            searchOptions=searchOptions
+            sortOptions=sortOptions
+            setOrder=(action 'setOrder')
+        }}
+        {{#discover.results
+            (html-attributes data-test-results=true)
+            as |result|
+        }}
+            {{registries-search-result result=result class=(local-class 'BrandedRegistriesSearchResult')}}
+        {{/discover.results}}
+
+        {{#unless totalResults}}
+            <div class='col-sm-8 col-xs-12'>
+                <div class='text-center text-muted'>
+                    <p class='lead'>
+                        {{t 'registries.discover.no_results'}}
+                    </p>
+                    {{t 'registries.discover.try_broadening'}}
+                </div>
+            </div>
+        {{/unless}}
+
+        {{#if (gt maxPage 1) }}
+            <div local-class='Pagination' class='col-xs-12'>
+                {{search-paginator
+                    current=searchOptions.page
+                    maximum=maxPage
+                    pageChanged=(action 'changePage')
+                }}
+            </div>
+        {{/if}}
+
+    {{/if}}
+
+{{/registries-discover-search}}
```

Comment:
Looking into DRYing this up.

Reply:
Hmm not quite sure how to DRY this up without getting too deep in the weeds. I tried creating a component `registries-discover-page` and then dumping everything below `registries-header` in there, but there is a lot of untangle in order to pass the correct properties through. There is a lot of logic in the controller.

Reply:
i'm not sure the best path, especially given time constraints...

we've used what we've been calling the "manager pattern" to isolate state management for some complex components, which would go something like:
- move controller state and all interaction with the `shareSearch` service into a plain class (e.g. [DraftRegistrationManager](https://github.com/CenterForOpenScience/ember-osf-web/blob/a6a936ff8ed5e0b3ce9be6844d1e20f5f0a14640/lib/registries/addon/drafts/draft/draft-registration-manager.ts)) or a component that has just a `yield` (no DOM) in its template (e.g. [SubjectsManager](https://github.com/CenterForOpenScience/ember-osf-web/blob/a6a936ff8ed5e0b3ce9be6844d1e20f5f0a14640/lib/osf-components/addon/components/subjects/manager/component.ts))
- share the manager (class instance or yielded hash) among the various components that make up the discover page
- non-manager components should:
  - treat the manager properties/fields/state as read-only (could potentially enforce this with types) and call the manager's actions to mutate
  - themselves hold only ephemeral state (e.g. should a dropdown be open or collapsed?), or state derived from manager state
- use the controller minimally, e.g. for syncing query params with the manager

does that sort of thing seem helpful and/or achievable-within-constraints, or is that the sort of too-deep-in-the-weeds approach you're trying to avoid?